### PR TITLE
feat(core): add ecdsa-binding encrypt flag

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
- this enables the optional 'ecdsa binding' feature in nanotdf containers

### CODE REVIEW

**Review by pflynn-virtru — DISMISSED** [review]

**Comment by jakedoublev** [comment]
This looks good. Can we please add an e2e test for this functionality to verify? https://github.com/opentdf/otdfctl/blob/main/tests/encrypt-decrypt.bats#L50

File: cmd/tdf-inspect.go [hunk 1]
```diff
@@ -36,36 +45,65 @@ func tdf_InspectCmd(cmd *cobra.Command, args []string) {
...
+				Method:                result.ZTDFManifest.Method,
+				IntegrityInformation:  result.ZTDFManifest.IntegrityInformation,
+				EncryptionInformation: result.ZTDFManifest.EncryptionInformation,
+			},
+			Attributes: result.Attributes,
+		}
 
-	c.PrintJson(m)
+		c.PrintJson(m)
+	} else if result.NanoHeader != nil {
```
Comment (jrschumacher) [inline cmd/tdf-inspect.go:80]: @dmihalcik-virtru this is great!! Thank you.

File: tests/encrypt-decrypt.bats [hunk 2]
```diff
@@ -53,6 +53,15 @@ teardown() {
   diff $INFILE_GO_MOD $RESULTFILE_GO_MOD
 }
 
+@test "roundtrip NANO, no attributes, file, ecdsa binding" {
+  ./otdfctl encrypt -o $OUTFILE_GO_MOD --host $HOST --tls-no-verify $DEBUG_LEVEL $WITH_CREDS --ecdsa-binding --tdf-type nano $INFILE_GO_MOD
+  ./otdfctl decrypt -o $RESULTFILE_GO_MOD --host $HOST --tls-no-verify $DEBUG_LEVEL $WITH_CREDS --tdf-type nano $OUTFILE_GO_MOD
+  diff $INFILE_GO_MOD $RESULTFILE_GO_MOD
+  ./otdfctl $HOST $WITH_CREDS inspect $OUTFILE_GO_MOD
+  ecdsa_enabled="$(./otdfctl $HOST $WITH_CREDS inspect $OUTFILE_GO_MOD | jq .ecdsaEnabled)"
```
Comment (jrschumacher) [inline tests/encrypt-decrypt.bats:61]:
```suggestion
  ecdsa_enabled="$(./otdfctl --host $HOST --tls-no-verify $WITH_CREDS inspect $OUTFILE_GO_MOD | jq .ecdsaEnabled)"
```

File: tests/encrypt-decrypt.bats [hunk 3]
```diff
@@ -53,6 +53,15 @@ teardown() {
   diff $INFILE_GO_MOD $RESULTFILE_GO_MOD
 }
 
+@test "roundtrip NANO, no attributes, file, ecdsa binding" {
+  ./otdfctl encrypt -o $OUTFILE_GO_MOD --host $HOST --tls-no-verify $DEBUG_LEVEL $WITH_CREDS --ecdsa-binding --tdf-type nano $INFILE_GO_MOD
+  ./otdfctl decrypt -o $RESULTFILE_GO_MOD --host $HOST --tls-no-verify $DEBUG_LEVEL $WITH_CREDS --tdf-type nano $OUTFILE_GO_MOD
+  diff $INFILE_GO_MOD $RESULTFILE_GO_MOD
+  ./otdfctl $HOST $WITH_CREDS inspect $OUTFILE_GO_MOD
```
Comment (jrschumacher) [inline tests/encrypt-decrypt.bats:60]:
```suggestion
  ./otdfctl --host $HOST --tls-no-verify $WITH_CREDS inspect $OUTFILE_GO_MOD
```

**Review by jrschumacher — CHANGES_REQUESTED** [review]
Tests failing due to missing flag string

**Review by jrschumacher — APPROVED** [review]

**Pull request merged**